src/processor/operator/persistent/reader/csv/csv_error_handler.cpp

```diff
@@ -228,27 +189,38 @@ void LocalCSVFileErrorHandler::handleError(BaseCSVReader* reader, const CSVError
         flushCachedErrors();
     }
 
-    ++linesPerBlock[error.blockIdx].invalidLines;
-    cachedErrors.push_back(error);
+    ++linesPerBlock[error.warningData.blockIdx].invalidLines;
+    if (cacheIgnoredErrors) {
+        cachedErrors.push_back(error);
+    }
 }
 
-void LocalCSVFileErrorHandler::reportFinishedBlock(BaseCSVReader* reader, uint64_t blockIdx,
-    uint64_t numRowsRead) {
-    sharedErrorHandler->reportFinishedBlock(reader, blockIdx, numRowsRead);
+void LocalCSVFileErrorHandler::reportFinishedBlock(uint64_t blockIdx, uint64_t numRowsRead) {
+    linesPerBlock[blockIdx].validLines += numRowsRead;
+    linesPerBlock[blockIdx].doneParsingBlock = true;
 }
```

Comment:
Line number data is now also cached locally in the error handler (without locks) and occassionally flushed to the shared error handler